### PR TITLE
Added some performance comments

### PR DESCRIPTION
**Pull request by RoosterDragon**
I've added some comments to help explain some parts of the code in performance sensitive areas - as well as to call out areas where the code needs to be fast to avoid people accidentally changing loops into LINQ and such in hot paths. I've prefixed such comments with `PERF:` for searchability.

The PR also includes a few commits with some minor cleanup.

### CODE REVIEW

**Comment by Mailaender** [comment]
:+1:

File: OpenRA.Game/Actor.cs [hunk 1]
```diff
@@ -206,6 +217,7 @@ public bool Equals(Actor other)
 
 		public override string ToString()
 		{
+			// PERF: Avoid format strings.
```
Comment (Phrohdoh) [inline OpenRA.Game/Actor.cs]: Question: is `string.Concat` any faster than `+` concatenation?

File: OpenRA.Game/Renderer.cs [hunk 2]
```diff
@@ -111,6 +111,7 @@ public void BeginFrame(int2 scroll, float zoom)
 
 		public void SetViewportParams(int2 scroll, float zoom)
 		{
+			// PERF: SetViewportParams is slow. Only call it when things change.
```
Comment (Phrohdoh) [inline OpenRA.Game/Renderer.cs]: This should go on the line above the method declaration.

Comment (RoosterDragon) [inline OpenRA.Game/Renderer.cs]: It applies to the individual `*Render.SetViewportParams` calls, not the method itself

File: OpenRA.Game/Actor.cs (hunk 1 above)
Comment (RoosterDragon) [inline OpenRA.Game/Actor.cs]: The compiler turns `+` into calls to `string.Concat` for you, handily.

File: OpenRA.Game/Renderer.cs (hunk 2 above)
Comment (Phrohdoh) [inline OpenRA.Game/Renderer.cs]: I see.
Yes please, though I do not know how to do so.

Comment (RoosterDragon) [inline OpenRA.Game/Renderer.cs]: Updated a bit.

**Comment by Phrohdoh** [comment]
:+1: looks good to me.

**Pull request merged**